packages/server/src/adapters/node-http/nodeHTTPRequestHandler.ts

```diff
@@ -28,9 +28,12 @@ export async function nodeHTTPRequestHandler<
       return await opts.createContext?.(opts);
     };
 
-    const query = opts.req.query
-      ? new URLSearchParams(opts.req.query as any)
-      : new URLSearchParams(opts.req.url!.split('?')[1]);
+    // a framework could pre-parse query in a way that could crash URLSearchParams
+    // invalidTrpcQuery can be added to a request object to let trpc know
+    const query: URLSearchParams =
+      opts.req.query && !opts.req.invalidTrpcQuery
```

Comment:
I'd name it by the action it'll perform, not by the "state" the request is in. Eg. `forceQueryFromUrl` or `forceUrlQuery` provides some meaning in itself, while `invalidTrpcQuery` doesn't tell you what will happen, but only indicates that something is "off".

Reply:
@kamilogorek I went with invalidTrpcQuery to differentiate the query that trpc uses internally from the query used by the framework. I'm happy to go with something like forceTrpcQueryFromUrl. My original thinking was that it leaks implementation details to the user that could change. For example, not all frameworks have a req.url property to make a query from although the ones I can think of already have a req.query parameter that is compatible with the trpc query and if they didn't, then they could just add a req.url property to generate the query from.